### PR TITLE
Interactive Video Reward Leave Dialog UI tests

### PR DESCRIPTION
**Pull request by me2k**
## JIRA
[AAG-2889]

## DESCRIPTION
Added Interactive Video Reward Leave Dialog UI tests and additional configuration options to the Test App.

I will create a ticket to repair the other broken UI Tests.

## SCREENSHOTS
Screenshot_20230328_142559

## TESTS
Screenshot 2023-03-28 at 14 17 38

[AAG-2889]: https://superawesomeltd.atlassian.net/browse/AAG-2889?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/src/androidTest/java/tv/superawesome/demoapp/VideoAdUITest.kt [hunk 1]
```diff
@@ -472,6 +469,56 @@ class VideoAdUITest {
         verifyUrlPathCalled("/vast/clickthrough")
     }
 
+    @Test
+    fun test_iv_video_warn_dialog_press_close() {
+        // Given VPAID Ad
+        val testData = TestData.videoVpaidPJ
+        stubIntents()
```
Comment (gunhansancar) [inline app/src/androidTest/java/tv/superawesome/demoapp/VideoAdUITest.kt:476]: Is there any outgoing click in this test which starts an Activity? Seems we don't need intent stubbing. Let's just check

File: app/src/androidTest/java/tv/superawesome/demoapp/VideoAdUITest.kt [hunk 2]
```diff
@@ -472,6 +469,56 @@ class VideoAdUITest {
...
+        CommonInteraction.clickItemAt(testData)
+
+        // When close is clicked
+        CommonInteraction.waitForCloseButtonThenClick()
+
+        VideoWarnInteraction.checkVisible()
+
+        VideoWarnInteraction.clickResume()
+
+        Thread.sleep(20000)
```
Comment (gunhansancar) [inline app/src/androidTest/java/tv/superawesome/demoapp/VideoAdUITest.kt:513]: This is work around :) I wonder if we can check the `subtitleView` appeared and the ad auto closes itself at the end? Also why do we wait for end of the video?

File: app/src/androidTest/java/tv/superawesome/demoapp/VideoAdUITest.kt [hunk 3]
```diff
@@ -472,6 +469,56 @@ class VideoAdUITest {
...
+
+        // Then
+        CommonInteraction.checkSubtitleContains("${testData.placement} adClosed")
+    }
+
+    @Test
+    fun test_iv_video_warn_dialog_press_resume() {
+        // Given VPAID Ad
+        val testData = TestData.videoVpaidPJ
+        stubIntents()
```
[the remark this answers is not in the document]
Comment (me2k) [reply app/src/androidTest/java/tv/superawesome/demoapp/VideoAdUITest.kt:498]: I'll remove it

File: app/src/androidTest/java/tv/superawesome/demoapp/VideoAdUITest.kt (hunk 2 above)
Comment (me2k) [reply app/src/androidTest/java/tv/superawesome/demoapp/VideoAdUITest.kt:513]: This is deliberate, I want to test that the video reaches the end and when you tap the close button it doesn't show the Video leave warning dialog if AdEnded has been received.

**Review by gunhansancar — APPROVED** [review]

**Pull request merged**